elasticsearch_collectd.py

```diff
@@ -1114,17 +1047,101 @@ def __str__(self):
+    at level debug, info, notice, warning, and error
+    to collectd's built in logging.  Suppresses extraneous
+    info and debug statements using a "verbose" boolean
+
+    Inherits from logging.Handler
+
+    Arguments
+        plugin -- name of the plugin (default 'unknown')
+        verbose -- enable/disable verbose messages (default False)
+    """
+    def __init__(self, plugin="unknown", verbose=False):
+        """Initializes CollectdLogHandler
+        Arguments
+            plugin -- string name of the plugin (default 'unknown')
+            verbose -- enable/disable verbose messages (default False)
+        """
+        self.verbose = verbose
+        self.plugin = plugin
+        logging.Handler.__init__(self, level=logging.NOTSET)
+
+    def emit(self, record):
+        """
+        Emits a log record to the appropraite collectd log function
+
+        Arguments
+        record -- str log record to be emitted
+        """
+        try:
+            if record.msg is not None:
+                if record.levelname == 'ERROR':
+                    collectd.error('%s : %s' % (self.plugin, record.msg))
+                elif record.levelname == 'WARNING':
+                    collectd.warning('%s : %s' % (self.plugin, record.msg))
+                elif record.levelname == 'NOTICE':
+                    collectd.notice('%s : %s' % (self.plugin, record.msg))
+                elif record.levelname == 'INFO' and self.verbose is True:
+                    collectd.info('%s : %s' % (self.plugin, record.msg))
+                elif record.levelname == 'DEBUG' and self.verbose is True:
+                    collectd.debug('%s : %s' % (self.plugin, record.msg))
+        except Exception as e:
+            collectd.warning(('{p} [ERROR]: Failed to write log statement due '
+                              'to: {e}').format(p=self.plugin,
+                                                e=e
+                                                ))
+
+
+class CollectdLogger(logging.Logger):
+    """Logs all collectd log levels via python's logging library
+    Custom python logger that forwards log statements at
+    level: debug, info, notice, warning, error
+
+    Inherits from logging.Logger
+
+    Arguments
+    name -- name of the logger
+    level -- log level to filter by
+    """
+    def __init__(self, name, level=logging.NOTSET):
+        """Initializes CollectdLogger
+
+        Arguments
+        name -- name of the logger
+        level -- log level to filter by
+        """
+        logging.Logger.__init__(self, name, level)
+        logging.addLevelName(25, 'NOTICE')
+
+    def notice(self, msg):
+        """Logs a 'NOTICE' level statement at level 25
+
+        Arguments
+        msg - log statement to be logged as 'NOTICE'
+        """
+        self.log(25, msg)
+
+
+# Set up logging
+logging.setLoggerClass(CollectdLogger)
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
```

Comment:
Because the python logger is doing 100% pass through to collectd's logging methods, we want the filter set to a very low level.  Collectd will filter the log levels itself.  So debug is the preferred setting.